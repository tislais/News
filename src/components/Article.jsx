// Article - Display an articles title, author, and description
import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{author}</p>
      <p>{description}</p>
    </>
  );
};

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Article;
