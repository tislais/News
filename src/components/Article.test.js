import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article.jsx';

describe('Article componenet', () => {
  it('renders article', () => {
    const article = {
      title: 'title',
      author: 'author',
      description: 'description'
    };

    const { asFragment } = render(
      <Article 
        title={article.title} 
        author={article.author} 
        description={article.description} 
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
