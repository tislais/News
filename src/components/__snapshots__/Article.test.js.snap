// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article componenet renders article 1`] = `
<DocumentFragment>
  <h1>
    title
  </h1>
  <p>
    author
  </p>
  <p>
    description
  </p>
</DocumentFragment>
`;
